keps/sig-scheduling/4832-async-preemption/README.md

```diff
@@ -0,0 +1,410 @@
+
+### Consideration to race condition
+
+Thanks to the nomination at `PostFilter`, this new asynchronous preemption shouldn't make any race condition between several scheduling cycles.
+
+Here, I'll discuss what happens in which scenario, and make sure there's no worry.
+
+Let's say pod1 is during the preemption process (node1) at the preemption goroutine, the next scheduling cycle is scheduling pod2. 
+
+#### The pod2's scheduling is successful (pod2 is equal or lower priority than pod1)
+
+As I described above, pod1's `PostFilter` nominates pod1 for node1.
+
+At the scheduling cycle, the scheduler takes such nominated pods that are equal or higher priority than pod1 into consideration;
+meaning, pod2 won't rob pod1 of the place on node1.
+
+#### The pod2's scheduling is successful (pod2 is higher priority than pod1)
+
+Even though pod1 is nominated for the node, the scheduler allows pod2 to take node1, where the pod1's preemption made the space. 
+
+Then, when pod1 comes back to the scheduling cycle, it may not be able to land on node1 because pod2 is scheduled there now.
+It happens with both the current and this KEP's scheduler, so no issue here.
+
+#### The pod2's scheduling is failed and starts the preemption (pod2 is equal or lower priority than pod1)
+
+The preemption also takes nominated Pods into consideration when calculating the preemption target.
+
+Therefore, if, coincidently, two preemptions for pod1 and pod2 select the same Node after all, 
+then the preemption for pod2 should decide to make the space for pod1 and pod2.
+
+So, we don't have to worry about two preemption targeting the same Node make any issue.
+
+#### The pod2's scheduling is failed and starts the preemption (pod2 is higher priority than pod1)
+
+The pod2's preemption ignores pod1's nomination for node1.
+
+If, coincidently, two preemptions for pod1 and pod2 select the same Node after all, 
+then the preemption for pod2 may just select the same preemption targets as pod1,
+and when pod1 comes back to the scheduling cycle, it (probably) cannot be scheduled on node1 because of pod2.
+
+But, this isn't an issue because the final result is completely the same as the current scheduler;
+with the current scheduler, pod1 preempts some Pods on node1, then pod2's scheduling starts, pod2 takes node1, 
+and when pod1 comes back to the scheduling cycle, it (probably) cannot be scheduled on node1 because of pod2.
+
+### Test Plan
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+##### Unit tests
+
+- `/pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go`: `2024-09-07` - `85.4`
+- `/pkg/scheduler/framework/preemption/preemption.go`: `2024-09-07` - `27.2`
```

Comment:
This seems fairly low. Can we increase the test coverage there first, before making changes for this KEP?